### DISCUSSION TITLE
Production Server not Running [v6]

### DISCUSSION THREAD

**Discussion by LouisRiverstone** [adonisjs/core#4579] [Help] [answered]
Hi guys, I'm having a problem running the server within Docker after building the application, I can't execute any command and it gives me the following error:

Cannot find package 'ts-node' imported from /app/ace.js
![image](https://github.com/adonisjs/core/assets/17641979/9c948345-2b9a-4cf7-9fd5-346bb86f17e7)

I even tried installing the assembler separately to try to run

Any listed way to solve it?

**Comment by RomainLanz** [comment]
Hey @LouisRiverstone! 👋🏻 

Can you please paste your `package.json` file?
The `ace` file should be rewritten when building to remove the import of `ts-node`.

**Comment by LouisRiverstone** [comment]
Here the packege-json of builded folder


{
  "name": "adonisjs-api-stater-kit",
  "version": "0.0.0",
  "private": true,
  "type": "module",
  "license": "UNLICENSED",
  "scripts": {
    "start": "node bin/server.js",
    "build": "node ace build",
    "dev": "node ace serve --watch",
    "test": "node ace test",
    "lint": "eslint .",
    "format": "prettier --write .",
    "typecheck": "tsc --noEmit"
  },
  "imports": {
    "#controllers/*": "./app/controllers/*.js",
    "#exceptions/*": "./app/exceptions/*.js",
    "#models/*": "./app/models/*.js",
    "#mails/*": "./app/mails/*.js",
    "#services/*": "./app/services/*.js",
    "#listeners/*": "./app/listeners/*.js",
    "#events/*": "./app/events/*.js",
    "#middleware/*": "./app/middleware/*.js",
    "#validators/*": "./app/validators/*.js",
    "#providers/*": "./providers/*.js",
    "#policies/*": "./app/policies/*.js",
    "#abilities/*": "./app/abilities/*.js",
    "#database/*": "./database/*.js",
    "#start/*": "./start/*.js",
    "#tests/*": "./tests/*.js",
    "#config/*": "./config/*.js",
    "#repositories/*": "./app/repositories/*.js",
    "#resources/*": "./app/resources/*.js",
    "#rules/*": "./app/rules/*.js",
    "#jobs/*": "./app/jobs/*.js",
    "#utils/*": "./app/utils/*.js"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^7.4.0",
    "@adonisjs/eslint-config": "^1.3.0",
    "@adonisjs/prettier-config": "^1.3.0",
    "@adonisjs/tsconfig": "^1.3.0",
    "@japa/api-client": "^2.0.3",
    "@japa/assert": "^3.0.0",
    "@japa/plugin-adonisjs": "^3.0.1",
    "@japa/runner": "^3.1.2",
    "@swc/core": "^1.4.14",
    "@types/luxon": "^3.4.2",
    "@types/node": "^20.12.7",
    "@types/sinon": "^17.0.3",
    "@types/uuid": "^9.0.8",
    "eslint": "^8.57.0",
    "prettier": "^3.2.5",
    "ts-node": "^10.9.2",
    "typescript": "^5.4.5"
  },
  "dependencies": {
    "@acidiney/bull-queue": "^1.5.1",
    "@adonisjs/auth": "^9.2.0",
    "@adonisjs/core": "^6.9.0",
    "@adonisjs/cors": "^2.2.1",
    "@adonisjs/drive": "^2.3.0",
    "@adonisjs/limiter": "^2.3.0",
    "@adonisjs/lucid": "^20.5.1",
    "@adonisjs/transmit": "^0.7.0",
    "@aws-sdk/client-s3": "^3.564.0",
    "@japa/expect": "^3.0.2",
    "@vinejs/vine": "^2.0.0",
    "file-chunked": "^1.0.4",
    "ioredis": "^5.4.1",
    "luxon": "^3.4.4",
    "mysql2": "^3.9.5",
    "pino-pretty": "^11.0.0",
    "reflect-metadata": "^0.2.2",
    "sinon": "^17.0.2",
    "uuid": "^9.0.1"
  },
  "eslintConfig": {
    "extends": "@adonisjs/eslint-config/app"
  },
  "prettier": "@adonisjs/prettier-config"
}

**Reply by RomainLanz** [reply]
I notice that you installed `uuid`, not that you can natively generate UUID in Node.js now.

```ts
import { randomUUID } from 'node:crypto'

randomUUID()
```

**Reply by LouisRiverstone** [reply]
Nice, i will replace uuid for node:crypto!

**Comment by RomainLanz** [comment]
Can you also share your Dockerfile to see how you build your application?

**Reply by LouisRiverstone** [reply]
Here my Dockerfile



`
FROM node:lts-alpine AS build

WORKDIR /build

RUN apk update

COPY . .

RUN [ "npm", "install"]

RUN [ "npm", "run", "build" ]

FROM node:lts-alpine

RUN apk update && \
  apk add \
  nodejs npm \
  coreutils \
  nano \
  git \
  supervisor

COPY --from=build /build /app

WORKDIR /app

RUN [ "npm", "install"]

RUN ["chmod", "+x", "/app/entrypoint.sh"]

ENTRYPOINT [ "./entrypoint.sh" ]
`

**Reply by LouisRiverstone** [reply, marked as answer]
I has removed build steps from Dockerfile, updated packages including "@swc/core": "^1.5.3" like last adonis realease plain installation, now everthing working fine! thanks for helping!
